compiler/rustc_resolve/src/access_levels.rs

```diff
@@ -127,99 +124,59 @@ impl<'r, 'a> AccessLevelsVisitor<'r, 'a> {
 
 impl<'r, 'ast> Visitor<'ast> for AccessLevelsVisitor<'ast, 'r> {
     fn visit_item(&mut self, item: &'ast ast::Item) {
-        let inherited_item_level = match item.kind {
+        let mut access_level = match item.vis.kind.is_pub() {
+            true => self.prev_level,
```

Comment:
The `prev_level` field is unnecessary and can be removed.
The same information can be obtained by using `self.r.access_levels.map.get(...)` on the parent `LocalDefId`.